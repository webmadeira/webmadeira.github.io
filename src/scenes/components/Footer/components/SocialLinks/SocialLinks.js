import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { H3 } from '../../../../../components/typography'
import facebookIcon from '../../../../../assets/img/icon/facebook.svg'
import instagramIcon from '../../../../../assets/img/icon/instagram.svg'
import linkedinIcon from '../../../../../assets/img/icon/linkedin.svg'
import twitterIcon from '../../../../../assets/img/icon/twitter.svg'

const socialIcons = {
  facebook: facebookIcon,
  instagram: instagramIcon,
  linkedin: linkedinIcon,
  twitter: twitterIcon,
}

const SocialList = styled.ul`
  list-style-type: none;
  display: flex;
  flex-direction: row;
  margin-top: 15px;
`

const SocialListItem = styled.li`
  width: 34px;
  height: 34px;
  margin-right: 16px;
`

const renderSocialLinks = links => links.map(link => (
  <SocialListItem key={link.id}>
    <a href={link.url} rel="noopener">
      <img src={socialIcons[link.name]} alt={link.name} />
    </a>
  </SocialListItem>
))

const SocialLinks = ({ links }) => (
  <div>
    <H3>SOCIAL</H3>
    <SocialList>
      {renderSocialLinks(links)}
    </SocialList>
  </div>
)

SocialLinks.propTypes = {
  links: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
  })).isRequired,
}

export default SocialLinks
