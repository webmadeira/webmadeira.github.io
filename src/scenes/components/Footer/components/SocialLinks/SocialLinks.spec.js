import React from 'react'
import { shallow } from 'enzyme'

import SocialLinks from './SocialLinks'

describe('SocialLinks component', () => {
  it('must render properly', () => {
    const links = [
      { id: 1, name: 'test social', url: '/testSocial' },
    ]

    const component = shallow(<SocialLinks links={links} />)

    expect(component).toMatchSnapshot()
  })
})
