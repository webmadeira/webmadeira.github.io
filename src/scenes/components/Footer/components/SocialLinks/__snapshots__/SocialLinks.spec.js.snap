// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLinks component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SocialLinks
    links={
      Array [
        Object {
          "id": 1,
          "name": "test social",
          "url": "/testSocial",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <styled.h3>
          SOCIAL
        </styled.h3>,
        <styled.ul>
          <styled.li>
            <a
              href="/testSocial"
              rel="noopener"
            >
              <img
                alt="test social"
                src={undefined}
              />
            </a>
          </styled.li>
        </styled.ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "SOCIAL",
        },
        "ref": null,
        "rendered": "SOCIAL",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.li>
              <a
                href="/testSocial"
                rel="noopener"
              >
                <img
                  alt="test social"
                  src={undefined}
                />
              </a>
            </styled.li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "children": <a
                href="/testSocial"
                rel="noopener"
              >
                <img
                  alt="test social"
                  src={undefined}
                />
              </a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="test social"
                  src={undefined}
                />,
                "href": "/testSocial",
                "rel": "noopener",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "test social",
                  "src": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "a",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <styled.h3>
            SOCIAL
          </styled.h3>,
          <styled.ul>
            <styled.li>
              <a
                href="/testSocial"
                rel="noopener"
              >
                <img
                  alt="test social"
                  src={undefined}
                />
              </a>
            </styled.li>
          </styled.ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "SOCIAL",
          },
          "ref": null,
          "rendered": "SOCIAL",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.li>
                <a
                  href="/testSocial"
                  rel="noopener"
                >
                  <img
                    alt="test social"
                    src={undefined}
                  />
                </a>
              </styled.li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "children": <a
                  href="/testSocial"
                  rel="noopener"
                >
                  <img
                    alt="test social"
                    src={undefined}
                  />
                </a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="test social"
                    src={undefined}
                  />,
                  "href": "/testSocial",
                  "rel": "noopener",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "test social",
                    "src": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "a",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
